File: gin-go-backend/internal/errors/error_map.go
package errors

import "errors"

func Normalize(err error) AppError {
	if err == nil {
		return InternalServerError("unknown error")
	}
	var ae AppError
	if errors.As(err, &ae) {
		return ae
	}
	return InternalServerError("an unexpected error occurred")
}

File: django-python-backend/apps/api/errors/error_map.py
from __future__ import annotations

from apps.api.errors.app_error import AppError
from apps.api.errors.http_errors import InternalServerError, ValidationError


def map_to_app_error(error: Exception) -> AppError:
    if isinstance(error, AppError):
        return error
    if isinstance(error, (ValueError, TypeError)):
        return ValidationError(detail=str(error))
    return InternalServerError(detail='An unexpected error occurred')
